src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.TSocket.cs

```diff
@@ -121,6 +142,17 @@ public override void Abort()
                 CancelWriteToSocket();
             }
 
+            public override async ValueTask DisposeAsync()
+            {
+                Transport.Input.Complete();
+                Transport.Output.Complete();
+
+                CompleteOutput(null);
```

Comment:
`CompleteOutput(null);` should be `CancelReadFromSocket();CancelWriteToSocket();`

Reply:
I wonder if you don't need something similar here: https://github.com/aspnet/AspNetCore/blob/5b2f3fb5f7f24ac3e91c5150a55cc411b2b36b76/src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvConnection.cs#L168-L171 to make sure `_processingTask` isn't stuck waiting for data to be read from/written to the socket.

Reply:
This is the problem: https://github.com/redhat-developer/kestrel-linux-transport/pull/80/files#r317943929